src/server.py

```diff
@@ -27,38 +35,130 @@ def server():
+
+        try:
+            uri = parse_request(message)
+        except HTTPException as exception:
+            msg = response_error(exception)
+        try:
+            file_data_tuple = resolve_uri(uri)
+        except HTTPException as exception:
+            msg = response_error(exception)
+        try:
+            msg = response_ok(file_data_tuple)
+        except HTTPException:
+            msg = (response_error("500 Internal Server Error"))
+
+        conn.sendall(msg.encode("utf-8"))
         conn.close()
 
 
-def response_ok():
-    """Function returns an HTTP '200 OK' response."""
-    response = (b"HTTP/1.1 200 OK\r\n"
-                b"Host: 127.0.0.1:5000\r\n\r\n"
-                b"Connection Successful")
-    return response
+def resolve_uri(uri):
+    if uri.find('../') is not -1:
+        raise HTTPException('403 Forbidden')
+    elif uri[-1] == '/':
+        return response_ok((generate_directory_html(uri), 'text/html'))
+    else:
+        try:
+            file_data_tuple = get_file_data(uri)
+        except HTTPException:
+            raise HTTPException('404 File Not Found')
+        return file_data_tuple
 
 
-def response_error():
-    """Function returns an HTTP '500 Internal Server Error' response."""
-    response = (b"HTTP/1.1 500 Internal Server Error\r\n"
-                b"Host: 127.0.0.1:5000\r\n\r\n"
-                b"Server Error")
-    return response
+def get_file_data(uri):
+    '''Reads file and returns a tuple: (file_dat, content-type).'''
+    try:
+        file_b = io.open(ROOT + '/' + uri, 'rb')
+    except IOError:
+        raise HTTPException('404 File Not Found')
+    mimetype = guess_type(uri)[0]
+    binary_file = file_b.read()
+    file_b.close
+    try:
+        unicode_file = binary_file.encode('utf8')
+    except AttributeError:
+        pass
+    return (unicode_file, mimetype)
+
+
+def generate_directory_html(directory):
+    """Generate an HTML string showing the contents of a directory.
 
+    os.listdir and ignoring hidden files function from:
+    from: http://stackoverflow.com/questions/7099290/how-to-ignore-hidden-
+    files-using-os-listdir-python
+    """
+    try:
+        file_list = []
+        for f in os.listdir(ROOT + directory):
+            if not f.startswith('.') and f is not None:
+                file_list.append(f)
+        file_list.sort()
+    except ValueError:
+        # need to do something else here
+        return HTTPException("404 File Not Found")
+    # except FileNotFoundError:
+    #     """FileNotFoundError seems to be an OSX thing, and this line catches it
+    #     despite the linter being very unhappy."""
+        return HTTPException("404 File Not Found")
+    out_string = '<html>\n\t<body>\n\t\t<ul>\n'
+    for f in file_list:
+        last_file = len(file_list) - 1
+        if f == file_list[last_file]:
+            out_string += '\t\t\t<li>{}</li>\n\t\t</ul>\n\t</body>\n</html>'\
+                          .format(f)
```

Comment:
could you avoid the conditional here by adding `<li>` items for all the files, and then once you've completed iterating over them all, adding the closing `</ul></body></html>`?